.github/workflows/magma-build-3rd-party.yml

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Magma Build & Publish 3rd Party Dependencies
+on:
+  push:
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+    branches:
+      - master
+      - 'v1.*'
+    paths:
+      - third_party/**
+      - .github/workflows/magma-build-3rd-party.yml
+  schedule:
+    - cron: 0 4 * * 0
+  workflow_dispatch:
+    inputs:
+      production:
+        description: Is production build?
+        type: boolean
+        default: false
+
+jobs:
+  build_dependencies:
+    runs-on: ubuntu-latest
+    container: ghcr.io/magma/magma/devcontainer:latest
+    env:
+      repository: debian-ci
+
+    strategy:
+      fail-fast: false
+      matrix:
+        distribution: [ focal ]
+        architecture: [ amd64, all ]
+        dependency:
+          - aioeventlet
+          - aioh2
+          - asn1c
+          - bcc
+          - folly
+          - freediameter
+          - gnutls
+          - grpc
+          - libfluid
+          - liblfds
+          - magma-cpp-redis
+          - magma-libtacopie
+          - nettle
+          - prometheus-cpp
+          - ryu
+          - sentry-native
+
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+
+      - name: Build debian package
+        working-directory: third_party/build
+        run: ./build.py --no-install ${{ matrix.dependency }}
+
+      - name: Test installability of the debian package built
+        working-directory: third_party/build
+        run: sudo apt-get install --yes --allow-downgrades ./*.deb
+
+      - run: echo "repository=debian-test" >> $GITHUB_ENV
+        if: github.ref == 'refs/heads/master'
+
+      - run: echo "repository=debian-prod" >> $GITHUB_ENV
+        if: inputs.production && contains('["maxhbr", "nstng", "jheidbrink", "lkreutzer", "Neudrino", "MoritzThomasHuebner", "tmdzk", "wolfseb"]', github.actor)
```

Comment:
Do we want to build and upload to production repositories? For some other artifacts (for example AGW Docker images, and I think the AGW Debian package), we instead promote existing artifacts as part of the release.

Reply:
This is just an option. One does not have to use it. However, it might be become handy during release.

Reply:
Refactored into a safeguard check.

Reply:
If we promote or not, is out of scope of the ticket!
(And does not really matter much, as one does not need to use the option. But having such a option and flexibility is good, from my experience.)